consensus.go

```diff
@@ -146,3 +147,16 @@ func (c *Consensus) ExternalIP() (net.IP, error) {
 	// we know it cannot be nil and is valid
 	return net.ParseIP(externalIP), nil
 }
+
+// UseIPProtocol will set the IP Protocol to use for http requests
+// to the sources. If zero, it will not discriminate. This is useful
+// when you want to get the external IP in a specific protocol.
+// Protocol only supports 0, 4 or 6.
+func (c *Consensus) UseIPProtocol(protocol uint) error {
```

Comment:
How about `UseNetwork(string)`, then the arguments are just `tcp4` or `tcp6`? That would mirror what people are used to with the `net` package, and it would make your Dial logic simpler, as you could just do:

```
if wantNetwork != "" && network != wantNetwork {
    // reject connection
}
```

Reply:
One potential downside is that if the net package gains tcp alternatives in the future, like QUIC, then go-external-ip users might have to adapt their network string.

If you want to keep the ipv4 vs ipv6 parameter, what I'd then suggest is to use `int` instead of `uint`, just because it's easier to use.